### PR TITLE
Update Nexus Allowlist container images

### PR DESCRIPTION
**Pull request by JimMadge**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :arrow_heading_up: Summary

Update Nexus Allowlist to a release which allows installing archived CRAN packages.

### :closed_umbrella: Related issues

Related to #2107

### CODE REVIEW

File: data_safe_haven/infrastructure/programs/sre/software_repositories.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def __init__(
                 container_group_name=f"{stack_name}-container-group-software-repositories",
                 containers=[
                     containerinstance.ContainerArgs(
-                        image="caddy:2.8.4",
+                        image="caddy:2.8",
```
Comment (jemrobinson) [inline data_safe_haven/infrastructure/programs/sre/software_repositories.py]: Why has this changed?

**Review by jemrobinson — CHANGES_REQUESTED** [review]
LGTM but are we sure we don't want this in the `release-v5.0.0` branch?

File: data_safe_haven/infrastructure/programs/sre/software_repositories.py (hunk 1 above)
Comment (JimMadge) [reply data_safe_haven/infrastructure/programs/sre/software_repositories.py]: Feels like an over specification to me.

**Comment by JimMadge** [comment]
> LGTM but are we sure we don't want this in the `release-v5.0.0` branch?

_I'm_ not very worried about it but it does change (quite significantly) what you can access through the CRAN proxy.

**Comment by jemrobinson** [comment]
OK, but we need what goes into v5.0.0 to be deployable. Either we need this fix or we need changes to the smoke tests so they actually work. I'm not sure which is better - what do you think?

**Comment by JimMadge** [comment]
> OK, but we need what goes into v5.0.0 to be deployable. Either we need this fix or we need changes to the smoke tests so they actually work. I'm not sure which is better - what do you think?

I think #2114 should fix the smoke tests without changing the access to packages.
I think it might just be that v5.0.0 has a known limitation that you can only install the latest version of each allowed CRAN package in T3. (I _think_ that must have been true in v4 too).

**Comment by jemrobinson** [comment]
Are you able to easily test whether #2114 actually *does* fix the smoke tests? If not, let me know and I'll try to deploy something.

File: data_safe_haven/infrastructure/programs/sre/software_repositories.py (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/infrastructure/programs/sre/software_repositories.py]: No, we need it. The automated Docker image updater looks for the latest release, not the latest minor release, so this change is made automatically.

**Comment by jemrobinson** [comment]
Sounds like we might want this to go into release 5.0.0. Am I OK to change the target of this PR @JimMadge ?

**Comment by JimMadge** [comment]
@craddm @jemrobinson are we happy to put this in for the release?

**Review by jemrobinson — APPROVED** [review]
LGTM

**Pull request merged**